ti/instrumentation/utils.go

```diff
@@ -434,6 +450,13 @@ func IsManualExecution(cfg *tiCfg.Cfg) bool {
 	return false
 }
 
+func IsPushTriggerExecution(cfg *tiCfg.Cfg) bool {
+	if (cfg.GetSourceBranch() == cfg.GetTargetBranch()) && !IsManualExecution(cfg) {
```

Comment:
Check source and target branch for empty strings.

Reply:
IsManualExecution takes care of that

Reply:
Flip the `&&` condition. In case of manual execution, both will be empty and you'll return true